### PR TITLE
Add Learner Support to our Sidebar

### LINKED ISSUE #229: Add Learner Support to our Sidebar
**Is your feature request related to a problem? Please describe.**
Time to start adding learner features! Let's start by displaying learner courses on the sidebar.

**Describe the solution you'd like**
Screen Shot 2022-10-12 at 11 19 30 PM

Currently, the API will return all courses a user belongs to, but a user can never join a course as a learner, and thus all of the courses that the API will return are contributor courses. I recommend using some test data to make sure that the Learner section displays the correct data.

1. Add a section for Learner courses. Highly encourage to reuse the components used to create the section for Contributor courses.
2. Create the section for "Search Courses." As of right now, the button doesn't have to link to another page.
3. Add the button for going to the Settings Page. As of right now, the button doesn't have to link to another page.
4. Change the Icon for each Contributor course to an icon that indicates whether the course is a public or private course. Refer to Jocelyn or Madison/Figma on what icons to use.

### PR DESCRIPTION
**Pull request by ashayp22**
## Description

- Added a `<Drawer.Tab/>` Stack list for Learner Courses on the Sidebar with a Custom React Navigation component.
- Removed User Email, Name, and Profile Picture from the Sidebar and replaced with a button for navigating to the Account Info page.
- Added a container for Searching for Courses.

Fixes #229 

## Screenshots

IMG_0787

### CODE REVIEW

File: client/src/navigator/Drawer/Drawer.js [hunk 1]
```diff
@@ -128,100 +132,188 @@ const generateTabs = (tabData) => tabData.map((element, index) => (
...
-                horizontal: 'never',
-              }}
+        <StyledButton
+          title="LEARNER"
+          fontSize={15}
+          variant="learner"
+          onPress={() => props.navigation.navigate('Apply', { from: 'HomeBaseCase' })}
+        />
+
+        <SplitDrawerItemList
```
Comment (21xiaofanli) [inline client/src/navigator/Drawer/Drawer.js:224]: got a console log warning that one of the props is an object instead of an array

File: client/src/navigator/Drawer/SplitDrawerItemList.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  secondRouteNames,
+  middleChildComponent,
+}) => {
+  const buildLink = useLinkBuilder()
+
+  const focusedRoute = state.routes[state.index]
+  const focusedDescriptor = descriptors[focusedRoute.key]
+  const focusedOptions = focusedDescriptor.options
+
+  console.log(focusedOptions)
```
Comment (21xiaofanli) [inline client/src/navigator/Drawer/SplitDrawerItemList.js]: why is this here tho

File: client/src/navigator/Drawer/SplitDrawerItemList.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        style={drawerItemStyle}
+        to={buildLink(route.name, route.params)}
+        onPress={onPress}
+      />
+    )
+  }
+
+  // Create the two Drawer.Item lists
+  const firstRoutesList = state.routes
+    .filter((obj) => firstRouteNames.includes(obj.name))
```
Comment (21xiaofanli) [inline client/src/navigator/Drawer/SplitDrawerItemList.js:117]: more specific names if this component wont be reused

File: client/src/navigator/Drawer/SplitDrawerItemList.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  console.log(focusedOptions)
+
+  /**
+   * Creates a DrawerItem given route data
+   * @param route = The route data
+   * @param i - The index of the route in the React Navigator
+   * @param zeroBasedShift - A shift to apply in the instance the route belongs to the second list
+   * @returns
+   */
+  const createCustomDrawerItem = (route, i, zeroBasedShift = 0) => {
```
Comment (21xiaofanli) [inline client/src/navigator/Drawer/SplitDrawerItemList.js:54]: do u need to default zerobasedshift to 0 if 1 out of the 2 instances already has the prop passed in as 0

**Review by 21xiaofanli — CHANGES_REQUESTED** [review]
i already changed a lot of my code, but this looks good besides the few comments i listed

File: client/src/navigator/Drawer/SplitDrawerItemList.js (hunk 3 above)
Comment (ashayp22) [reply client/src/navigator/Drawer/SplitDrawerItemList.js:117]: I'm going to keep the general name in the event we reuse the component in the future.
